### PR TITLE
Fix corner-case kamino out of bound accesses

### PR DESCRIPTION
**Pull request by Guirec-Maloisel**
## Description

While tracking a bug (that I didn't end up finding) I ran the Kamino internal unit tests with wp.config.mode set to "debug" and wp.config.verify_cuda set to True, and came across a few CUDA errors:
- One in a contact unit test providing a single capacity for a 3-world model when allocating contacts, eventually dereferencing world_max_contacts out of bounds
- One in a reset unit test modifying a joint from fixed to free in a builder without ensuring that dynamics properties (a_j, b_j...) are extended to 6-arrays (one per dof), eventually dereferencing these out of bounds
- One in a contact unit test with warmstart, because of a missing bound in a loop in contact matching that dereferences the sorted contact keys out of bounds when no match is found on the last key

This concerns mostly corner cases, but is still worth fixing

## Checklist

- [ ] New or existing tests cover these changes
- [ ] The documentation is up to date with these changes
- [ ] `CHANGELOG.md` has been updated (if user-facing change)

## Test plan

The Kamino-internal unit tests now run with debug/verify_cuda (can be set manually in setup_tests() in tests/\_\_init\_\_.py). For convenience I also gated a few graph captures to ensure they don't run with verify_cuda (as this is not allowed).

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved contact warm-starting safety by adding bounds checks to avoid out-of-range access.
  * Fixed floating-base joint initialization when using modified DoF setup.
  * Updated heterogeneous contact capacity allocation to match per-world sizing.
* **Performance & Compatibility**
  * Disabled CUDA graph capture/execution when CUDA verification is enabled (applies across examples and tests).
* **Documentation**
  * Rephrased test documentation for joint-state reset validation messaging.

### CODE REVIEW

**Review by chschuma-disney — COMMENTED** [review]
Looks good to me, except what coderabbit pointed out, which does make sense to me.

File: newton/_src/solvers/kamino/_src/solvers/warmstart.py [hunk 1]
```diff
@@ -225,6 +225,8 @@ def _warmstart_contacts_by_matched_geom_pair_key_and_position(
 
         # Update the current old-key to check in the next iteration
         k += 1
+        if start + k >= sorted_contact_keys_old.shape[0]:
+            break
         old_key = sorted_contact_keys_old[start + k]
```
[the remark this answers is not in the document]
Comment (Guirec-Maloisel) [reply newton/_src/solvers/kamino/_src/solvers/warmstart.py]: I think that the while condition target_key == old_key and the sentinel values used beyond active contacts would prevent this from happening in practice. But I agree that your suggestion is cleaner as a loop bound, so I updated as suggested.

**Review by chschuma-disney — APPROVED** [review]

**Comment by adenzler-nvidia** [comment]
I think one of them is even making the CI flaky at the moment, so lets get this merged!
